### PR TITLE
Contact page and buttons

### PR DESCRIPTION
**Pull request by filipvvv**
## What does this PR do and why?

Add a contact page and buttons which direct to that page. Upon submission, the contact page sends a message in the JuiceboxDAO Discord server via web request.

## Screenshots or screen recordings

https://user-images.githubusercontent.com/96905094/215924562-98dd5971-6cf1-46ca-98ff-bbdbb8d89e4b.mp4

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [x] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [x] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by filipvvv** [comment]
@wraeth-eth would love your help getting the `Select` at line 54 of `src/pages/contact/Contact.tsx` to be the same height as `Input` at line 94. See the video if interested!

Seems like it needs some type of antd override, but not sure where to get started.

**Comment by wraeth-eth** [comment]
> @wraeth-eth would love your help getting the `Select` at line 54 of `src/pages/contact/Contact.tsx` to be the same height as `Input` at line 94. See the video if interested!
> 
> Seems like it needs some type of antd override, but not sure where to get started.

@filipvvv can take a look at this

**Comment by filipvvv** [comment]
@wraeth-eth had some time today and fixed the issue. It had to do with an override which sets `margin-bottom` on `Form.Item`. I was using the addOnAfter prop on an `Input` and the margins were combined due to some weird CSS interactions. Fixed with a simple `mb-0`.

This PR should be ready for review. I'd like to start by trialing it as-is, and I think adding more CTAs which direct to this page would be a good idea in the future. Could eventually integrate this with some type of CRM if STVG starts using one.

Goal is to increase conversion rates for people just visiting the page — even if they don't have Discord.

File: src/lib/discord.ts
Comment (tomquirk) [reply src/lib/discord.ts:13]: **suggestion (non-blocking)**: can also be nice here to have a single object parameter, vs 5 individual parameters. Or, have the required parameters as their own params, but optional ones in an object. E.g.

`createContactMessage(message, { name, subject, etc })`

Non blocking, whatever you think!

File: src/pages/contact/Contact.tsx
Comment (tomquirk) [inline src/pages/contact/Contact.tsx:36]: **suggestion**: might be worth a `console.error(e)` here to help debug exceptions if they arise.

File: src/pages/contact/Contact.tsx
Comment (tomquirk) [inline src/pages/contact/Contact.tsx:52]: **praise** nice idea!

**Review by tomquirk — CHANGES_REQUESTED** [review]
@filipvvv absolutely love this. Awesome job figuring out Next pages and API handlers. Just left a few minor comments but on the whole, looks great 🏓

File: src/pages/contact/Contact.tsx
Comment (tomquirk) [inline src/pages/contact/Contact.tsx:120]: **suggestion** this could use a `required` prop i think.

File: src/pages/contact/Contact.tsx
Comment (tomquirk) [inline src/pages/contact/Contact.tsx:131]: **suggestion** if we make the `message` a required field, then we can remove this disabled state and just let the user click `submit` to trigger the validation. Typically a nicer UX, because the user explicitly knows what they need to fix.

File: src/pages/contact/Contact.tsx
Comment (strath-m) [reply src/pages/contact/Contact.tsx:52]: will these be the placeholders for the contact form @filipvvv ?

**Comment by strath-m** [comment]
Overall love this idea and I think it could work really well, down to test it and see how we go, if people are using it we should make it a more prominent feature of jbm.

The only thing I would say from a visual perspective is to make the form field styling consistent with what we use in other areas of the UI (see create flow for examples).

Screen Shot 2023-02-07 at 9 11 00 am

Screen Shot 2023-02-07 at 9 10 49 am

Also, I would remove the divider between the description and the form fields, I think it adds an unnecessary separation.
Screen Shot 2023-02-07 at 9 21 49 am

File: src/lib/discord.ts
Comment (filipvvv) [reply src/lib/discord.ts:4]: Thank you! I'm used to just using `fetch` but axios is great.

File: src/lib/discord.ts
Comment (filipvvv) [reply src/lib/discord.ts:54]: Totally! Yes, this is the webhook URL. Changed to a variable for now. Might be worth setting as a Vercel environmental variable at some point.

File: src/lib/discord.ts
Comment (filipvvv) [reply src/lib/discord.ts:13]: Only validating message body was intentional — I could imagine somebody wanting to leave anonymous feedback, for example. Might be unlikely but I don't think leaving the option open is harmful. This is true on the contact page form as well — you can submit only once you type a message, but nothing else is needed.

The single object parameter is a great idea. Making this change now, and also making the create message function better handle empty fields.

File: src/pages/contact/Contact.tsx
Comment (filipvvv) [reply src/pages/contact/Contact.tsx:36]: Great idea, adding.

File: src/pages/contact/Contact.tsx
Comment (filipvvv) [reply src/pages/contact/Contact.tsx:52]: @strath-m yes — when a user selects a platform to contact them on, a different placeholder appears in the format appropriate to that platform.

File: src/pages/contact/Contact.tsx
Comment (filipvvv) [reply src/pages/contact/Contact.tsx:120]: Great idea, adding.

File: src/pages/contact/Contact.tsx
Comment (filipvvv) [reply src/pages/contact/Contact.tsx:131]: Sounds good to me! Replacing with antd form `required` logic.

File: src/lib/discord.ts
Comment (filipvvv) [reply src/lib/discord.ts:1]: Thank you!

**Comment by filipvvv** [comment]
@wraeth-eth had another weird CSS interaction with some strange border properties. Seems like it could be fixed by removing some overrides on the antd input addon, but don't want to mess up something elsewhere in the app. If you have some time would love your help — otherwise I can get back to it tomorrow.

@tomquirk and @strath-m — thank you both for fantastic feedback. I've implemented your suggestions; looking forward to wrapping this one up!

**Comment by tomquirk** [comment]
@filipvvv awesome thanks for considering the feedback! Heads up, just pushed a rebase with main and resolved conflicts.

This should be good to go. Will merge when pipeline suceeeds 🚀

**Review by tomquirk — APPROVED** [review]

**Comment by filipvvv** [comment]
@aeolianeth don't merge just yet — still have a CSS bug to fix, will do so tomorrow!

**Review by tomquirk — APPROVED** [review]

**Comment by tomquirk** [comment]
> @aeolianeth don't merge just yet — still have a CSS bug to fix, will do so tomorrow!

@filipvvv got it, lemme know!

**Comment by filipvvv** [comment]
Working well now. Dev was having an issue but it seems to be fixed on the Goerli preview. Appreciate your help — feel free to merge @tomquirk!

**Comment by filipvvv** [comment]
Never mind — seems a later commit of mine was overwritten. Fixing.

**Comment by wraeth-eth** [comment]
Sorry @filipvvv , this one got lost on my todos. let me know if you need any help with it. Maybe shoot me a Discord either dm or tag on a thread! 🙏

**Comment by tomquirk** [comment]
@filipvvv awesome, will fix conflict, have a final test and then merge if all good <3

**Comment by strath-m** [comment]
Screen Shot 2023-02-10 at 11 06 18 am

**Important:** Can this be moved below the 'Next' button before merge. Super awkward here and takes away from the primary CTA of the page.

**Comment by tomquirk** [comment]
@filipvvv will send back to you to address strath feedback above 🙏

**Review by tomquirk — APPROVED** [review]

**Comment by filipvvv** [comment]
There is still a minor issue with the border color on the `addOnAfter`, but not blocking IMO. Feel free to merge. Thanks for feedback and thanks again for bearing with me — last few days have been eventful.

**Comment by wraeth-eth** [comment]
Thanks @filipvvv !

**Pull request merged**
